consumer/pdata/common.go

```diff
@@ -293,8 +297,30 @@ func (a AttributeValue) Equal(av AttributeValue) bool {
 		return v.IntValue == av.orig.GetIntValue()
 	case *otlpcommon.AnyValue_DoubleValue:
 		return v.DoubleValue == av.orig.GetDoubleValue()
+	case *otlpcommon.AnyValue_ArrayValue:
+		vv := v.ArrayValue.GetValues()
+		avv := av.orig.GetArrayValue().GetValues()
+		if len(vv) != len(avv) {
+			return false
+		}
+
+		for i, val := range avv {
+			val := val
+			av := newAttributeValue(&vv[i])
+
+			switch av.orig.Value.(type) {
+			case *otlpcommon.AnyValue_ArrayValue, *otlpcommon.AnyValue_KvlistValue:
+				return false
+			}
```

Comment:
a simple if instead of switch?

Reply:
Also why returning false here? Can you document?

Reply:
False is a way to skip recursive check. I thought that we stick to the specification, in which the array value must be a scalar type. This prevented me from thinking about possible self-references as the array element and other similar problems

Reply:
You can still change this in a simple if statement.

Reply:
Idk if you meant the if that I've done